internal/services/monitor/monitor_scheduled_query_rules_alert_resource.go

```diff
@@ -395,17 +396,20 @@ func expandMonitorScheduledQueryRulesAlertingAction(d *pluginsdk.ResourceData) *
 	alertAction := expandMonitorScheduledQueryRulesAlertAction(alertActionRaw)
 	severityRaw := d.Get("severity").(int)
 	severity := strconv.Itoa(severityRaw)
-	throttling := d.Get("throttling").(int)
+	throttling, throttlingOk := d.GetOk("throttling")
 
 	triggerRaw := d.Get("trigger").([]interface{})
 	trigger := expandMonitorScheduledQueryRulesAlertTrigger(triggerRaw)
 
 	action := insights.AlertingAction{
-		AznsAction:      alertAction,
-		Severity:        insights.AlertSeverity(severity),
-		ThrottlingInMin: utils.Int32(int32(throttling)),
-		Trigger:         trigger,
-		OdataType:       insights.OdataTypeBasicActionOdataTypeMicrosoftWindowsAzureManagementMonitoringAlertsModelsMicrosoftAppInsightsNexusDataContractsResourcesScheduledQueryRulesAlertingAction,
+		AznsAction: alertAction,
+		Severity:   insights.AlertSeverity(severity),
+		Trigger:    trigger,
+		OdataType:  insights.OdataTypeBasicActionOdataTypeMicrosoftWindowsAzureManagementMonitoringAlertsModelsMicrosoftAppInsightsNexusDataContractsResourcesScheduledQueryRulesAlertingAction,
+	}
+
+	if throttlingOk {
```

Comment:
maybe this can happen simpler, can we just check for `throttling > 0` here? then you don't need to change the Schema and you can remove throttlingOk

Reply:
We could handle the default case where throttling is 0 here instead of setting the default to `nil` in the schema.
```suggestion
	if throttling, ok := d.Get("throttling").(int); ok && throttling != 0 {
		action.ThrottlingInMin = utils.Int32(int32(throttling))
	}
```